plugins/pulp_rpm/plugins/distributors/yum/publish.py

```diff
@@ -566,9 +568,23 @@ def initialize(self):
         self.context = UpdateinfoXMLFileContext(self.get_working_dir(), checksum_type,
                                                 self.get_conduit())
         self.context.initialize()
+
+        repo_id = self.get_repo().id
+        rpm_content_type = models.RPM._content_type_id.default
+        nevra_fields = ('name', 'epoch', 'version', 'release', 'arch')
+        unit_ids_list = list(repo_controller.get_associated_unit_ids(repo_id, rpm_content_type))
+        repo_nevra_q = models.RPM.objects.filter(id__in=unit_ids_list).scalar(*nevra_fields)
```

Comment:
This is a good idea. The line needed a subtle adjustment. Look for it in the next push.

`nevra_scalars = itertools.chain(*[q.scalar(*nevra_fields) for q in querysets])`